README.md

```diff
@@ -109,56 +109,80 @@ With IBM Cloud Satellite, you can bring your own compute infrastructure that is
+| `storage-class.yaml` | The list of vendor provided storage classes. Example [`storage-class.yaml`](https://github.com/IBM/ibm-satellite-storage/blob/master/config-templates/netapp/netapp-ontap-san/20.07/storage-class.yaml). |File | No |
+| `storage-class-template.yaml` | The storage class template. This template allows user create additioal storage classes on a Satellite cluster. The additional storage classes are created from the  parameters mentioned in `storage-class-parameters.json` Example [`storage-class-template.yaml`](https://github.com/IBM/ibm-satellite-storage/blob/master/config-templates/netapp/netapp-trident/20.07/storage-class-template.yaml). |File | No |
+| `storage-class-parameters.json` | The list of storage class parameters. User / admin can override the values from GUI. The parameter values are injected in storage class template to generate the storage class specifications to deploy on the target clusters. [`storage-class-parameters.json`](https://github.com/IBM/ibm-satellite-storage/blob/master/config-templates/netapp/netapp-trident/20.07/storage-class-parameters.json) |File | No |
+| `README.md` | Contains prerequisites, deployment steps, and additional information about your template. Copy and fill out the [README_TEMPLATE.md](/.github/README_TEMPLATE.md). |File | No |
+| `metadata.json` | This file contains the metadata for GUI display. Example [`metadata.json`](https://github.com/IBM/ibm-satellite-storage/blob/master/config-templates/netapp/netapp-trident/20.07/metadata.json). |File | No |
+
+
+### Defining Custom parameter reference
+The customer parameter has following attributes 
+| Attribute name | Description | Required | Default|
+| --- | --- | ---| --- |
+| `description` | A description of the parameter. | Yes| |
+| `displayname` | The display name of the parameter that is used in the Satellite UI. |Yes |
+| `name` | The name of the parameter in string format. | Yes | 
+| `default` | Optional. The default value for the parameter. If the does not specify the parameter in their configuration, this value entered as `default` is used. | No | Empty String|
+| `type` | Specify the parameter type. Valid parameter types are: `text`, `csv`, `number`, `boolean` | No | `text`|
+| `catagory` | Specify the parameter category `config` or `secret`. On UI , the `config` parameters will display  in `User Config Parameter` section and `secret` parameters will dislay in `User secrete paramater` section  | No | `config`|
+| `required` | Specify `true` or `false`. | No | `false`|
```

Comment:
I think it will be better to mention that when the default value for parameter is set, then set `required` to false.

Reply:
I disagree with this. In the UI, if they have a default value they can still delete it. If they delete it and its "supposed" to be required but follows this pattern, they could have an issue without realizing.